extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -39,7 +39,13 @@ public void set(T state) {
             Context context = Vertx.currentContext();
             if (context != null && VertxContext.isDuplicatedContext(context)) {
                 VertxContextSafetyToggle.setContextSafe(context, true);
```

Comment:
@Sanne @mkouba I've already changed internally this via https://github.com/quarkusio/quarkus/pull/30024, but still...maybe it could be just avoided?

if `context.getLocal(LOCAL_KEY) == state` do we need to call `setContextSafe`? are they indipendent?